dev/com.ibm.ws.kernel.boot_fat/fat/src/com/ibm/ws/kernel/service/ServerEndpointControlMBeanTest.java

```diff
@@ -69,172 +77,363 @@
+            mbean.pause(null);
+            fail("Didn't get expected exception from attempting to pause a server with an invalid target list");
 
         } catch (MBeanException e) {
         }
-        assertNotNull(server.waitForStringInLog("CWWKE0935W", logWaitTimeout));
+        assertNotNull("Didn't find CWWKE0931W in logs as expected.", server.waitForStringInLog("CWWKE0931W"));
 
         Log.exiting(c, METHOD_NAME);
     }
 
     /**
-     * Tests that the expected exception is thrown if the MBean resume method is issued with an invalid target.
+     * Tests that the expected exception is thrown if the MBean resume method is issued when no pauseable components exist
      *
      * @throws Exception
      */
+
+    @AllowedFFDC(PAUSEABLE_EXCEPTION_CLASS)
     @Test
-    public void testListEndpoints() throws Exception {
-        final String METHOD_NAME = "testListEndpoints";
+    public void testResumeAllNoPauseableComponents() throws Exception {
+        final String METHOD_NAME = "testResumeAllNoPauseableComponents";
         Log.entering(c, METHOD_NAME);
+        boolean noException = true;
+        restoreSavedConfig = true;
 
-        List<String> eps = mbean.listEndpoints();
-        assertTrue("Expected 0 endpoints from listEndpoints, got " + eps.size(), eps.size() == 0);
-
+        // remove the servlet feature so that all endpoints are deactivated including the default instance of httpEndpoint
+        server.setMarkToEndOfLog();
+        server.changeFeatures(Arrays.asList("localConnector-1.0", "endpointControl-1.0"));
+        assertNotNull("Didn't get expected config update log messages", server.waitForConfigUpdateInLogUsingMark(null, true));
+        try {
+            server.setMarkToEndOfLog();
+            mbean.resume();
+        } catch (MBeanException e) {
+            // based on the design of the PauseableComponent service, you may get an exception if no endpoint has ever been registered
+            // and pause or resume all is called on it or simply an info message that the operation completed if an endpoint has ever been registered
+            // but may be deactivated
+            Log.info(c, METHOD_NAME, "Caught expected exception: " + e.getCause());
+            noException = false;
+        }
+        // if we got an exception or the completion log message was found, test passed
+        final String res = server.waitForStringInLog("CWWKE0939I");
+        if (noException && ("".equals(res) || res == null)) {
+            fail("Failed to get either an exception or completion log message from attempting to resume an endpoint with no pauseable components");
+        }
         Log.exiting(c, METHOD_NAME);
     }
 
     /**
-     * Tests that the expected exception is thrown if the MBean resume method is issued with an invalid target.
+     * Tests that the expected exception is thrown if the MBean pause method is issued with invalid target list when no pauseable components exist
```

Comment:
This description isn't correct- this test is when resume is issued with an invalid/empty target list (but a pauseable component does exist)